nuxeo-connect-client/src/test/java/org/nuxeo/connect/connector/http/TestConnectHttpConnector.java

```diff
@@ -0,0 +1,477 @@
+
+    private static final int PORT = 17488;
+
+    private ConnectHttpConnector httpConnector;
+
+    public class CustomTestRequestHandler extends AbstractHandler {
+
+        private String expectedTargetSuffix = "";
+
+        private int expectedResponseStatus = HttpServletResponse.SC_OK;
+
+        private String expectedJSONResponse = "{}";
+
+        private boolean expectGzipHeader = true;
+
+        public void setExpectGzipHeader(boolean expectGzipHeader) {
+            this.expectGzipHeader = expectGzipHeader;
+        }
+
+        public void setExpectedTargetSuffix(String expectedTargetSuffix) {
+            this.expectedTargetSuffix = expectedTargetSuffix;
+        }
+
+        public void setExpectedResponseStatus(int expectedResponseStatus) {
+            this.expectedResponseStatus = expectedResponseStatus;
+        }
+
+        public void setExpectedJSONResponse(String expectedJSONResponse) {
+            this.expectedJSONResponse = expectedJSONResponse;
+        }
+
+        @Override
+        public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
+                throws IOException, ServletException {
+            assertThat(target.endsWith("/" + expectedTargetSuffix)).as("Target url %s should end with %s", target,
+                    expectedTargetSuffix).isTrue();
+            if (expectGzipHeader) {
+                assertThat(request.getHeader("Accept-Encoding")).contains("gzip");
+            }
+            response.setStatus(expectedResponseStatus);
+            try (ServletOutputStream os = response.getOutputStream()) {
+                os.print(expectedJSONResponse);
+            }
+        }
+
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        // Configure and start Jetty server
+        server = new Server();
+        ServerConnector connector = new ServerConnector(server);
+        connector.setHost(HOST);
+        connector.setPort(PORT);
+        connector.setIdleTimeout(60 * 1000); // 60 seconds
+        server.addConnector(connector);
+        // Enable gzip compression
+        GzipHandler gzipHandler = new GzipHandler();
+        gzipHandler.setMinGzipSize(0);
+        gzipHandler.setHandler(new CustomTestRequestHandler());
+        server.setHandler(gzipHandler);
+        server.start();
+
+        // Configure httpConnector
+        httpConnector = new ConnectHttpConnector();
+        httpConnector.overrideUrl = "http://" + HOST + ":" + PORT + "/" + BASE_RESOURCE + "/";
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        httpConnector.flushCache();
+        // remove potentially loaded DownloadingPackages
+        for (DownloadingPackage downloadingPackage : NuxeoConnectClient.getDownloadManager().listDownloadingPackages()) {
+            NuxeoConnectClient.getDownloadManager().removeDownloadingPackage(downloadingPackage.getId());
+        }
+        server.stop();
+        server.destroy();
```

Comment:
Do the `server` cleanup in a `finally` block as we really want to release the server port even if the `for` loop has an error for some reason.